### DISCUSSION TITLE
One possible Solution for S3

### DISCUSSION THREAD

**Discussion by pjmichel** [moosefs/moosefs#397] [Show and tell]
Hi all, since 1 year i use this simple setup for Sharing MFS over Minio and so S3, with multiple Goofys as S3 clients .

on mfsmount part i use this param :
mfsmaster=xxx
mfssubfolder=xxx
mfswritecachesize=2048
#optimisation lecture
mfsreadaheadsize=256
mfsreadaheadleng=4194304
mfsreadaheadtrigger=8388608

i mount it in a forder /mnt/san
over that with bindfs, i mount /mnt/nas over /mnt/san 

i think, that do fuse to temp hard disk to fuse IO 

and now by using minio gateway nas --quiet /mnt/nas &

i have minio that work as expect, so speed to browe the mfs file and no hang . 

I share and wrote multiple To with this setup without crash 
Hope that can help, surely need to advance testing, but could lead to a way of sharing via S3 without headackes

PS i use that simple way in the past this rsynd, for having my repo directly on mfs, with the same success

**Comment by footlooseboss** [comment]
Thank you for sharing. 

I’m curious to know if you’re still using this solution, and whether you have any remote nodes in your network.
